crates/typst-pdf/src/page.rs

```diff
@@ -626,18 +626,64 @@ fn write_text(ctx: &mut PageContext, pos: Point, text: &TextItem) {
         glyph_set.entry(g.id).or_insert_with(|| segment.into());
     }
     let fill_transform = ctx.state.transforms(Size::zero(), pos);
-    ctx.set_fill(&text.fill, true, fill_transform);
     if let Some(stroke) = &text.stroke {
+        ctx.content.begin_text();
         ctx.set_stroke(stroke, true, fill_transform);
         ctx.content
-            .set_text_rendering_mode(pdf_writer::types::TextRenderingMode::FillStroke);
+            .set_text_rendering_mode(pdf_writer::types::TextRenderingMode::Stroke);
+        ctx.set_font(&text.font, text.size);
+        ctx.set_opacities(text.stroke.as_ref(), None);
```

Comment:
Does the order between `ctx.set_xxx` `ctx.content.begin_text` `ctx.content.set_text...` matter?